docs/content/en/docs/implementing/tasks/_index.md

```diff
@@ -5,12 +5,72 @@ weight: 90
 hidechildren: false # this flag hides all sub-pages in the sidebar-multicard.html
 ---
 
-Keptn tasks are defined in a
+A
 [KeptnTaskDefinition](../../yaml-crd-ref/taskdefinition.md/)
-resource.
+resource defines tasks that the Keptn Lifecycle Toolkit runs
+as part of the pre- and post-deployment phases of a
+[KeptnApp](../../yaml-crd-ref/app.md) or
+[KeptnWorkload](../../concepts/workloads.md).
+
+A Keptn task runs as an application
+[container](https://kubernetes.io/docs/concepts/containers/),
+which runs as part of a Kubernetes
+[job](https://kubernetes.io/docs/concepts/workloads/controllers/job/).
 A task definition includes a function
 that defines the action taken by that task.
-It can be configured in one of three different ways:
+
+To implement a Keptn task:
+
+- Define a
+  [KeptnTaskDefinition](../yaml-crd-ref/taskdefinition.md)
+  resource that defines the container
+- [Annotate your workloads](../integrate/#annotate-workloads)
+  to integrate your task with Kubernetes
+- Add your task to the  [KeptnApp](../yaml-crd-ref/app.md)
+  resource that associates your `KeptnTaskDefinition`
+  with the pre- and post-deployment tasks that should run in it
+
+This page provides information to help you create your tasks:
+
+- Code your task in a [container](#containers)
```

Comment:
Maybe this is a better terminology, but I am not 100% happy 🤔 I would like to not use "container" here to avoid confusion with "container-runtime"
```suggestion
- Code your task in a [runner](#containers)
```

Reply:
What about "Code your task to use an appropriate [runner](https://docs.gitlab.com/runner/executors/kubernetes.html#how-the-runner-creates-kubernetes-pods)" ?

Reply:
Sounds good 👍

Reply:
done